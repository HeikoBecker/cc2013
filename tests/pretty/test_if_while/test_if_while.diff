1,21c1,4
< void (blub())
< {
< 	if ((a + b)) {
< 		;
< 	} else
< 		;
< 	while ((a[9])) {
< 		;
< 	}
< 	if ((((a->b).c) = (7 = (a || b))))
< 		if ((a ? b : c), (a = (a ? b : 7))) {
< 		} else {
< 			while ((7 ? (a = (b ? (a[9]) : (a || b))) : ((z && a) || (b + d)))) {
< 				goto b;
< 				continue;
< 				break;
< 				return;
< 			}
< 		}
< 	return (a ? b, (a[8]), (a || c), (a && (((a[9]) + (1->b)) - (d.c))) : ((a < b) != (a < b)));
< }
---
> test_if_while.c:25:16: error: Second and third operand of conditional operator have unsuiting types *int and int
> 		while(7? a = b ? c[9] : a || b : e && a || b + d) {
> 		             ^^^^
> 1 error(s)
