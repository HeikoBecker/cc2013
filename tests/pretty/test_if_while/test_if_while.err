test_if_while.c:25:16: error: Second and third operand of conditional operator have unsuiting types *int and int
		while(7? a = b ? c[9] : a || b : e && a || b + d) {
		             ^^^^
1 error(s)
