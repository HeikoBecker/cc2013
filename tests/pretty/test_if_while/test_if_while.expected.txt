struct G
{
	int c;
};

struct S
{
	struct G b;
};

void (blub(void))
{
	int a;
	int b;
	int (*c);
	int e;
	struct S (*d);
	struct G f;
	if ((a + b)) {
		;
	} else
		;
	while ((c[9])) {
		;
	}
	if ((((d->b).c) = ((7 == a) || b)))
		if ((a ? b : e)) {
			(a = (a ? b : 7));
		} else {
			while ((7 ? (a = (b ? (c[9]) : (a || b))) : ((e && a) || (b + d)))) {
				goto b;
				continue;
				break;
				return;
			}
		}
	return;
}
